### PR TITLE
Cut v0.4.1

### PR DESCRIPTION
**Pull request by astoycos**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -97,9 +98,47 @@ a PR. This must go through the regular PR review process and get merged into the
...
 
+Additionally you can navigate to the github release page and draft a release
+generating release notes there as well.
+
 ### Release Steps
 
-The following steps must be done by one of the [bpfman maintainers][bpfman-team]:
+The following steps must be done by one of the [bpfman maintainers][bpfman-team],
+please always try and cut a release candidate before pushing an official major
+or minor release:
```
Comment (Billy99) [inline RELEASE.md:108]:
```suggestion
or minor release.
```

File: RELEASE.md [hunk 2]
```diff
@@ -97,9 +98,47 @@ a PR. This must go through the regular PR review process and get merged into the
...
+
+For a **RELEASE CANDIDATE** release:
```
Comment (Billy99) [inline RELEASE.md:110]:
```suggestion
#### RELEASE CANDIDATE Release
```

File: RELEASE.md [hunk 3]
```diff
@@ -97,9 +98,47 @@ a PR. This must go through the regular PR review process and get merged into the
...
+    - Updates the bpfman-operator version in it's [Makefile](https://github.com/bpfman/bpfman/blob/main/bpfman-operator/Makefile):
+        - `VERSION ?= x.x.x`
+- Make sure CI is green and merge the update PR.
+- Create a tag using the `HEAD` of the `main` branch. This can be done using the `git` CLI or
+  Github's [release][release] page.
+- Tag the release using the commit on `main` where the changelog update merged.
+  This can  be done using the `git` CLI or Github's [release][release]
+  page.
+
+Once these steps are completed:
```
Comment (Billy99) [inline RELEASE.md:135]:
```suggestion
Once these steps are completed:

```
Add newline

File: RELEASE.md [hunk 4]
```diff
@@ -97,9 +98,47 @@ a PR. This must go through the regular PR review process and get merged into the
...
+- Make sure a new RPM has been built and pushed to the bpfman COPR repository.
+- Make sure all [actions][gh-actions] have completed successfully
+- Ensure images are built and updated with the new version tag at:
+  - quay.io/bpfman
```
Comment (Billy99) [inline RELEASE.md:139]:
```suggestion
    - quay.io/bpfman
```
Each quay.io image line needs 2 more leading space

File: RELEASE.md [hunk 5]
```diff
@@ -97,9 +98,47 @@ a PR. This must go through the regular PR review process and get merged into the
...
+  - quay.io/bpfman-bytecode
+  - quay.io/bpfman-userspace
 
 For a **PATCH** release:
```
Comment (Billy99) [inline RELEASE.md:143]:
```suggestion
#### PATCH Release
```

File: RELEASE.md [hunk 6]
```diff
@@ -110,13 +149,14 @@ For a **PATCH** release:
   `git checkout -b release-x.x.x <major.minor.patch>`
 - Create a pull request of the `<githubuser>/release-x.x.x` branch into the `release-x.x` branch upstream.
   Add a hold on this PR waiting for at least one maintainer/codeowner to provide a `lgtm`. This PR should:
```
Comment (Billy99) [inline RELEASE.md:151]:
```suggestion
  Add a hold on this PR waiting for at least one maintainer/codeowner to provide a `lgtm`. This PR:
```

File: RELEASE.md [hunk 7]
```diff
@@ -110,13 +149,14 @@ For a **PATCH** release:
...
+    - Updates the [Cargo.toml](https://github.com/bpfman/bpfman/blob/main/Cargo.toml) version for the workspace:
         - `version = "x.x.x"`
         - `bpfman = { version = "x.x.x", path = "./bpfman" }"`
         - `bpfman-api = { version = "x.x.x", path = "./bpfman-api" }`
         - Note: `bpfman-csi` does not need to be updated.
-    - Update the bpfman-operator version in it's [Makefile](https://github.com/bpfman/bpfman/blob/main/bpfman-operator/Makefile):
+    - Runs `cargo generate-lockfile`
+    - Updates the bpfman-operator version in it's [Makefile](https://github.com/bpfman/bpfman/blob/main/bpfman-operator/Makefile):
         - `VERSION ?= x.x.x`
     - Run `make bundle` from the bpfman-operator directory to update the bundle version.
```
Comment (Billy99) [inline RELEASE.md:161]:
```suggestion
    - Runs `make bundle` from the bpfman-operator directory to update the bundle version.
```

File: RELEASE.md [hunk 8]
```diff
@@ -141,15 +190,16 @@ For a **MAJOR** or **MINOR** release:
```
Comment (Billy99) [inline RELEASE.md:190]:
```suggestion
#### MAJOR or MINOR Release
```

File: bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml [hunk 9]
```diff
@@ -5,4 +5,4 @@ kind: Kustomization
 images:
 - name: quay.io/bpfman/bpfman-operator
   newName: quay.io/bpfman/bpfman-operator
-  newTag: latest
+  newTag: v0.4.1
```
Comment (Billy99) [inline bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml:8]: Was this on purpose?

File: examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+# Patch the bytecode.yaml to change tag on the "url" field (which is an
+# image) to new value. This actually overwrites the image with the same value.
+patches:
+  - target:
+      kind: UprobeProgram
+      name: go-uprobe-counter-example
```
Comment (Billy99) [inline examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml:9]:
```suggestion
      name: go-uretprobe-counter-example
```
I think all the Program type are uretprobe, right?

File: examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    patch: |-
+      - op: replace
+        path: "/spec/bytecode/image/url"
+        value: quay.io/bpfman-bytecode/go-uprobe-counter:v0.4.1
```
Comment (Billy99) [inline examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml:13]:
```suggestion
        value: quay.io/bpfman-bytecode/go-uretprobe-counter:v0.4.1
```

File: examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+# Patch the deployment.yaml to change container image in Daemonset
+# to new tag on the image.
+images:
+  - name: quay.io/bpfman-userspace/go-uprobe-counter
```
Comment (Billy99) [inline examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml:17]:
```suggestion
  - name: quay.io/bpfman-userspace/go-uretprobe-counter
```

File: examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    newName: quay.io/bpfman-userspace/go-uprobe-counter
```
Comment (Billy99) [inline examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml:18]:
```suggestion
    newName: quay.io/bpfman-userspace/go-uretprobe-counter
```

File: examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    newTag: v0.4.1
+resources: [../../selinux/go-uprobe-counter]
```
Comment (Billy99) [inline examples/config/v0.4.1-selinux/go-uretprobe-counter/kustomization.yaml:20]:
```suggestion
resources: [../../selinux/go-uretprobe-counter]
```

File: examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml [hunk 15]
```diff
@@ -0,0 +1,20 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+# Patch the bytecode.yaml to change tag on the "url" field (which is an
+# image) to new value. This actually overwrites the image with the same value.
+patches:
+  - target:
+      kind: UprobeProgram
+      name: go-uprobe-counter-example
```
Comment (Billy99) [inline examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml:9]:
```suggestion
      name: go-uretprobe-counter-example
```

File: examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+    patch: |-
+      - op: replace
+        path: "/spec/bytecode/image/url"
+        value: quay.io/bpfman-bytecode/go-uprobe-counter:v0.4.1
```
Comment (Billy99) [inline examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml:13]:
```suggestion
        value: quay.io/bpfman-bytecode/go-uretprobe-counter:v0.4.1
```

File: examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+# Patch the deployment.yaml to change container image in Daemonset
+# to new tag on the image.
+images:
+  - name: quay.io/bpfman-userspace/go-uprobe-counter
```
Comment (Billy99) [inline examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml:17]:
```suggestion
  - name: quay.io/bpfman-userspace/go-uretprobe-counter
```

File: examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+    newName: quay.io/bpfman-userspace/go-uprobe-counter
```
Comment (Billy99) [inline examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml:18]:
```suggestion
    newName: quay.io/bpfman-userspace/go-uretprobe-counter
```

File: examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+    newTag: v0.4.1
+resources: [../../base/go-uprobe-counter]
```
Comment (Billy99) [inline examples/config/v0.4.1/go-uretprobe-counter/kustomization.yaml:20]:
```suggestion
resources: [../../base/go-uretprobe-counter]
```

File: examples/config/v0.4.1/go-xdp-counter-sharing-map/kustomization.yaml [hunk 20]
```diff
@@ -0,0 +1,20 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+# Patch the bytecode.yaml to change tag on the "url" field (which is an
+# image) to new value. This actually overwrites the image with the same value.
+patches:
+  - target:
+      kind: XdpProgram
+      name: go-xdp-counter-example
```
Comment (Billy99) [inline examples/config/v0.4.1/go-xdp-counter-sharing-map/kustomization.yaml:9]:
```suggestion
      name: go-xdp-counter-sharing-map-example
```

File: examples/config/v0.4.1-selinux/go-xdp-counter-sharing-map/kustomization.yaml [hunk 21]
```diff
@@ -0,0 +1,20 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+# Patch the bytecode.yaml to change tag on the "url" field (which is an
+# image) to new value. This actually overwrites the image with the same value.
+patches:
+  - target:
+      kind: XdpProgram
+      name: go-xdp-counter-example
```
Comment (Billy99) [inline examples/config/v0.4.1-selinux/go-xdp-counter-sharing-map/kustomization.yaml:9]:
```suggestion
      name: go-xdp-counter-sharing-map-example
```

**Review by Billy99 — COMMENTED** [review]
Double check me on the yaml names with uretprobe and xdp sharing-map.

File: bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml (hunk 9 above)
Comment (astoycos) [reply bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml:8]: Nope good catch, maintaining these manifests are kind of a PITA :)

**Review by Billy99 — APPROVED** [review]
LGTM

Added a Hold label, @astoycos remove when you are ready for it to merge.

**Pull request merged**